xnmt/encoder.py

```diff
@@ -95,8 +113,8 @@ def transduce(self, sent):
       sent = module.transduce(sent)
     return sent
 
-  def get_train_test_components(self):
-    return self.modules
-
-
+  @decorators.recursive
```

Comment:
There will be many tedious recursion code because, for example the HTMLReportable need to recurse through all the members that implements the method html_report. By marking the code with this decorator, it does that automatically.